mergify_engine/queue/merge_train.py

```diff
@@ -66,6 +67,53 @@ class WaitingPull(typing.NamedTuple):
     config: queue.QueueConfig
 
 
+class TrainCarState(enum.Enum):
+    pending = "pending"
+    created = "created"
+    updated = "updated"
+    failed = "failed"
+
+
+json.register_type(TrainCarState)
```

Comment:
Enum a particularly cumbersome to use, especially with json.
Isn't a `typing.Literal` good enough?